spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DateFormatStringChecker.java

```diff
@@ -0,0 +1,118 @@
+package edu.umd.cs.findbugs.detect;
+
+import org.apache.bcel.Const;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.StringAnnotation;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DateFormatStringChecker extends OpcodeStackDetector {
+
+    private static final List<String> AM_PM_HOURS_FLAG_1 = new ArrayList<>(Arrays.asList("a", "h"));
+    private static final List<String> AM_PM_HOURS_FLAG_2 = new ArrayList<>(Arrays.asList("a", "K"));
+    private static final List<String> MILITARY_HOURS_FLAG_1 = new ArrayList<>(Arrays.asList(null,"H","a"));
+    private static final List<String> MILITARY_HOURS_FLAG_2 = new ArrayList<>(Arrays.asList(null,"k","a"));
+    private static final List<String> WEEK_YEAR_FLAG_1 = new ArrayList<>(Arrays.asList("w","Y","M","d"));
+
+    private static final List<Integer> CONST_ARRAY_LIST = new ArrayList<>(Arrays.asList((int)Const.INVOKESPECIAL,
+            (int)Const.INVOKEVIRTUAL, (int)Const.INVOKESTATIC, (int)Const.INVOKEINTERFACE));
+
+    private static final String SIG_CONSTANT_OPERAND_1 = "Ljava/lang/String;)V";
+    private static final String SIG_CONSTANT_OPERAND_2 = "Ljava/lang/String;Ljava/util/Locale;)V";
+    private static final String SIG_CONSTANT_OPERAND_3 = "Ljava/lang/String;Ljava/text/DateFormatSymbols;)V";
+    private static final String CLASS_CONSTANT_OPERAND = "java/text/SimpleDateFormat";
+    private static final String NAME_CONSTANT_OPERAND_1 = "<init>";
+    private static final String NAME_CONSTANT_OPERAND_2 = "applyPattern";
+    private static final String NAME_CONSTANT_OPERAND_3 = "applyLocalizedPattern";
+
+    private static final String BUG_TYPE = "FS_BAD_DATE_FORMAT_FLAG_COMBO";
+
+    String dateFormatString;
```

Comment:
This field is used only by one method, then better to replace it with a local variable.